pkg/realizer/workload/component.go

```diff
@@ -44,7 +44,7 @@ func NewResourceRealizer(workload *v1alpha1.Workload, repo repository.Repository
 	}
 }
 
-func (r *resourceRealizer) Do(ctx context.Context, resource *v1alpha1.SupplyChainResource, supplyChainName string, outputs Outputs) (*unstructured.Unstructured, *templates.Output, error) {
+func (r *resourceRealizer) Do(ctx context.Context, resource *v1alpha1.SupplyChainResource, supplyChainName string, supplyChainParams []v1alpha1.OverridableParam, outputs Outputs) (*unstructured.Unstructured, *templates.Output, error) {
```

Comment:
The supply-chain params may more properly belong in the instantiation of the resourceRealizer, as the one set is used in every iteration over the different resources.